server/src-lib/Hasura/Events/Lib.hs

```diff
@@ -345,10 +345,16 @@ decodeHeader logenv headerInfos (hdrName, hdrVal)
      decodeBS = TE.decodeUtf8With TE.lenientDecode
 
 addDefaultHeaders :: [HTTP.Header] -> [HTTP.Header]
-addDefaultHeaders hdrs = hdrs ++
-  [ (CI.mk "Content-Type", "application/json")
-  , (CI.mk "User-Agent", "hasura-graphql-engine/" <> T.encodeUtf8 currentVersion)
-  ]
+addDefaultHeaders hdrs = defaultHeaders <> rmDefaultHeaders hdrs
```

Comment:
@lexi-lambda This change has led me to believe that we have HTTP clients which are not using the Hasura.HTTP library basically (event trigger module and remote schema module):

The only modules which seem to be using this are:

Telemetry
Auth
Auth/JWT
CheckUpdates

Is it fine for EventTriggers and RemoteSchemas to maintain their own clients because these might be specific to these modules.

Reply:
This function appears to prioritize default headers over the headers that were passed in, but the code in `GraphQL.Execute` appears to prioritize the headers that were passed in over the default headers. Is the difference intentional?

Also, can this code at least share the `defaultHeaders` declaration with the one in `Hasura.HTTP`? Having it duplicated in both places seems worth avoiding if we can help it.

Reply:
This was my question. Is it worth sharing these tiny functions, when the bulk of the client is specified per module itself? Shouldn't HTTP.Hasura give out a client with these baked in for standard usage? Those who are not using it may do their own thing including defaultHeaders?

Reply:
I don’t know, as I haven’t looked at the code in detail, but my gut instinct is that

  1. these modules should probably be using `Hasura.HTTP`,

  2. but if they aren’t, and you don’t want to do the refactoring to make them use it right now, sharing the default headers is an alright first step.

If the first assumption isn’t true, then you could be right, but I’d be surprised if it were. Is there a reason these modules shouldn’t be using the same library?

Reply:
I don't think there is any real reason except the library didn't have proper functions to use with Event Triggers. I guess we should make an attempt to consolidate the library then.

Cool, I am just going to share the functions then. Let's do the full consolidation in a separate no-op refactor PR.

Reply:
Sounds good. 👍